paddle/fluid/framework/ir/mkldnn/params_to_int8_pass_tester.cc

```diff
@@ -0,0 +1,318 @@
+    op->SetAttr("data_format", std::string{"NCHW"});
+    op->SetAttr("dilations", std::vector<int>({1, 1}));
+    op->SetAttr("paddings", std::vector<int>({1, 1}));
+    op->SetAttr("strides", std::vector<int>({1, 1}));
+    return op;
+  }
+
+ protected:
+  void CreateProgram() override {
+    OpDesc* op = CreateBasicConvOp();
+    AddInput(op, "Input", input);
+    AddInput(op, "Filter", filter)->SetPersistable(true);
+    AddOutput(op, "Output", output);
+
+    op->SetAttr("Scale_weights", scale_weights);
+    op->SetAttr("Scale_in", 1.0f);
+    op->SetAttr("groups", groups);
+
+    if (HasBias()) {
+      AddInput(op, "Bias", bias);
+      op->SetAttr("Bias_scales", scale_bias);
+    }
+  }
+
+  void CheckOp(const OpDesc& op) const override {
+    CheckFilter(op);
+    if (HasBias()) {
+      CheckBias(op);
+    }
+  }
+
+  bool HasBias() const { return !bias.getData().empty(); }
+
+  void CheckFilter(const OpDesc& op) const {
+    EXPECT_EQ(op.GetAttrIfExists<std::vector<float>>("Scale_weights"),
+              std::vector<float>(1, 1));
+
+    auto filter_inputs = op.Input("Filter");
+    ASSERT_EQ(filter_inputs.size(), 1ul);
+
+    auto tensor = test_scope.GetTensor(filter_inputs[0]);
+    ASSERT_EQ(tensor.dtype(), phi::DataType::INT8);
+
+    auto filter_ptr = tensor.data<int8_t>();
+    ASSERT_NE(filter_ptr, nullptr);
+    auto length = tensor.numel() / scale_weights.size();
+    for (int64_t i = 0; i < tensor.numel(); i++) {
+      EXPECT_EQ(filter_ptr[i],
+                static_cast<int8_t>(std::round(filter.getData()[i] *
+                                               scale_weights[i / length])));
+    }
+  }
+
+  void CheckBias(const OpDesc& op) const {
+    EXPECT_EQ(op.GetAttrIfExists<std::vector<float>>("Bias_scales"),
+              std::vector<float>(1, 1));
+
+    auto bias_inputs = op.Input("Bias");
+    ASSERT_EQ(bias_inputs.size(), 1ul);
+
+    auto tensor = test_scope.GetTensor(bias_inputs[0]);
+    auto bias_ptr = tensor.data<int32_t>();
+    ASSERT_NE(bias_ptr, nullptr);
+    auto length = tensor.numel() / scale_bias.size();
+    for (int64_t i = 0; i < tensor.numel(); i++) {
+      EXPECT_EQ(bias_ptr[i], static_cast<int32_t>(std::round(
+                                 bias.getData()[i] * scale_bias[i / length])));
+    }
+  }
+
+ private:
+  const Data input;
+  const Data filter;
+  const Data output;
+  const std::vector<float> scale_weights;
+  const int groups;
+
+  const Data bias;
+  const std::vector<float> scale_bias;
+};
+
+struct ParamsToInt8PassTest {
+  void RunPassTest() {
+    auto graph = program->CreateGraph();
+
+    auto pass = PassRegistry::Instance().Get("params_to_int8_pass");
+    graph.reset(pass->Apply(graph.release()));
+
+    program->CheckGraph(graph);
+  }
+  std::unique_ptr<ProgramStrategy> program;
+};
+
+Data GenericInput() { return Data({1, 4, 1, 1}, {1.5f, 1.5f, 1.5f, 1.5f}); }
+Data GenericOutput() { return GenericInput(); }
+
+TEST(ParamToInt8Pass, conv_without_bias_o1i1h1w1) {
```

Comment:
You could think of using parametrized tests (maybe for `Data`?), as there are some nice options and would probably make code shorter, but on the other hand it would get pretty messy with all the namings

Reply:
I tried  to parametrize by shared_ptr<ConvProgramStrategy> but it got completely unreadable